src/material.js

```diff
@@ -174,6 +191,19 @@ export class Material {
         return new Lattice(this.lattice);
     }
 
+    getInchiStringForHash() {
+        const derivedProperties = this.getDerivedProperties();
+        if (derivedProperties.length > 0) {
+            const inchi = derivedProperties ? this.getDerivedPropertyByName('inchi') : null;
+            if (inchi) {
+                return inchi.value
+            } else {
+                throw new Error("Hash cannot be created. Missing InChI string in derivedProperties")
```

Comment:
Let's move this to the end of the checks if inchi cannot be extracted, we should not be calling this function